server/filestore.go

```diff
@@ -3462,6 +3521,35 @@ func (fs *fileStore) LoadLastMsg(subject string) (subj string, seq uint64, hdr,
 	return sm.subj, sm.seq, sm.hdr, sm.msg, sm.ts, nil
 }
 
+// LoadNextMsg will find the next message matching the filter subject starting at the start sequence.
+// The filter subject can be a wildcard.
+func (fs *fileStore) LoadNextMsg(filter string, wc bool, start uint64) (subj string, seq uint64, hdr, msg []byte, ts int64, err error) {
+	fs.mu.RLock()
+	defer fs.mu.RUnlock()
+
+	if fs.closed {
+		return _EMPTY_, 0, nil, nil, 0, ErrStoreClosed
+	}
+	if start < fs.state.FirstSeq {
+		start = fs.state.FirstSeq
+	}
+
+	for _, mb := range fs.blks {
+		// Skip blocks that are less than our starting sequence.
+		if start > atomic.LoadUint64(&mb.last.seq) {
+			continue
+		}
+		if sm, expireOk, err := mb.firstMatching(filter, wc, start); err == nil {
```

Comment:
So if `err != nil`, we don't care really what the error is, we will fall through and return `ErrStoreEOF`, right?

Reply:
Correct, start will select the mb, but we may not match anything in there. If so we simply move on. Should we return an error if not MsgNotFound?

Reply:
Depends on the caller, but really MsgNotFound could be equivalent on what you return now, which is EOF because technically, it is as if we were going through the end of the file to look for it and we didn't find it.

Reply:
Correct, what I meant was a non msg not found error should we return that, like corrupt msg etc?

Reply:
Yes, it would help debug if someday there is an error reading the file, etc..

Reply:
ok fixed.